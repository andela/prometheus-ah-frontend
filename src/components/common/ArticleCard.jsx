import React from 'react';
import moment from 'moment';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

const ArticleCard = ({ article }) => (
  <div className="card shadow-lg border-0 mb-5 featured-post">
    <div className="card-body">
      <Link to={`/articles/${article.slug}`} className="card-title post-title">
        <h3 className="card-title post-title">
          {article.title}
        </h3>
      </Link>
      <p className="card-text post-date">
        {moment(article.createdAt).fromNow()}
        <span className="ml-4 font-weight-light font-italic">
          {article.Tags ? article.Tags[0].name.toUpperCase() : '' }
        </span>
      </p>
      <div className="card-text post-description">
        {article.body.length < 200 ? article.body : `${article.body.substring(0, 200)}...`}
      </div>
      <br />
      <div className="row">
        <div className="col-sm-6 author">
          {`${article.User.firstname} ${article.User.lastname}`}
          <br />
          <p>
            <small className="text-muted font-italic">
              {article.readingTime}
            </small>
          </p>
        </div>
      </div>
    </div>
  </div>
);

ArticleCard.propTypes = {
  article: PropTypes.shape({}).isRequired,
};

export default ArticleCard;
