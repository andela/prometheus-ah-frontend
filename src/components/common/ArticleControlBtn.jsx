import React from 'react';
import { Link } from 'react-router-dom';

const ArticleControlBtn = (props) => {
  const {
    link, slug, handleDelete // eslint-disable-line
  } = props;
  return (
    <div>
      <Link
        to={link}
        className="dropdown-item"
      >
  Edit
      </Link>
      <a
        className="dropdown-item"
        onClick={() => handleDelete(slug)}
        href="#"
      >
  Delete
      </a>
    </div>
  );
};

const ReportBtn = (props) => {
  const { showReportModal } = props; // eslint-disable-line
  return (
    <a
      className="dropdown-item"
      onClick={showReportModal}
      href="#"
    >
  Report
    </a>
  );
};

export { ArticleControlBtn, ReportBtn };
