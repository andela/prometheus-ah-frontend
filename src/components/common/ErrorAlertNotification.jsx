import React from 'react';

/**
   * Error alert notification
   * @param {*} props - Response object
   * @returns {*} props.errors - with props.errors
   */
const ErrorAlertNotification = props => (
  <div className="alert alert-danger fade show" role="alert">
    <button type="button" className="close" onClick={() => props.onClick()}>
      <span aria-hidden="true">&times;</span>
    </button>
    <strong>{props.errors}</strong>
  </div>);

export default ErrorAlertNotification;
