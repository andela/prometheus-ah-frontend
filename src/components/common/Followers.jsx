import React from 'react';
import PropTypes from 'prop-types';

const Followers = ({ followers }) => (
  <div className="modal fade" id="followers" tabIndex="-1" role="dialog" aria-labelledby="exampleModalLabel" aria-hidden="true">
    <div className="modal-dialog modal-dialog-centered" role="document">
      <div className="modal-content">
        <div className="modal-header">
          <h5 className="modal-title" id="exampleModalLabel">Followers</h5>
          <button type="button" className="close" data-dismiss="modal" aria-label="Close">
            <span aria-hidden="true">&times;</span>
          </button>
        </div>
        <div className="modal-body">
        followers
        </div>
      </div>
    </div>
  </div>
);

Followers.propTypes = {
  users: PropTypes.shape({}),
};

export default Followers;
