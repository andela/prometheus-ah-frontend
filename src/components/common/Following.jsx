import React from 'react';
import PropTypes from 'prop-types';

const Following = ({ following }) => (
  <div className="modal fade" id="following" tabIndex="-1" role="dialog" aria-labelledby="exampleModalLabel" aria-hidden="true">
    <div className="modal-dialog modal-dialog-centered" role="document">
      <div className="modal-content">
        <div className="modal-header">
          <h5 className="modal-title" id="exampleModalLabel">Following</h5>
          <button type="button" className="close" data-dismiss="modal" aria-label="Close">
            <span aria-hidden="true">&times;</span>
          </button>
        </div>
        <div className="modal-body">
          { `${following.authorsIFollow}` }
        </div>
      </div>
    </div>
  </div>
);

Following.propTypes = {
  users: PropTypes.shape({}),
};

export default Following;
