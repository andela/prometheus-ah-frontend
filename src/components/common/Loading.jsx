import React from 'react';
import loader from '../../images/loader.gif';

const Loading = () => (
  <div className="loading-center">
    <img src={loader} className="text" alt="Loading..." />
  </div>
);

export default Loading;
