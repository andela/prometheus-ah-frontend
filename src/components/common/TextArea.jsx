import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

const TextArea = ({
  field, value, error, onChange
}) => (
  <div className={classnames({ 'text-danger': error })}>
    <textarea
      name={field}
      value={value}
      onChange={onChange}
      className="form-control"
    />
    { error && <p className="text-danger">{error}</p>}
  </div>
);

TextArea.propTypes = {
  field: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  error: PropTypes.shape({
  }),
  onChange: PropTypes.func.isRequired,
};


export default TextArea;
