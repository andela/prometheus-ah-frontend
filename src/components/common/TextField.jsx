import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

const TextField = ({
  field, value, error, type, onChange, id, placeholder
}) => (
  <div className={classnames({ 'text-danger': error })}>
    <input 
      className="form-control"
      type={type}
      name={field}
      value={value}
      onChange={onChange}
      placeholder={placeholder}
      id={id}
      required
      />
    { error && <p className="text-danger">{error}</p>}
  </div>
);

TextField.propTypes = {
  field: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  error: PropTypes.array,
  type: PropTypes.string.isRequired,
  placeholder: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  id: PropTypes.string.isRequired
};

TextField.defaultProps = {
  type: 'text'
};


export default TextField;
