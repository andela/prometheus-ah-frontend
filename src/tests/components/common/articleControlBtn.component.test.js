import React from 'react';
import { shallow } from 'enzyme';
import { ArticleControlBtn, ReportBtn } from '../../../components/common/ArticleControlBtn';

describe('renders component correctly', () => {
  it('should render ArticleControlBtn', () => {
    const wrapper = shallow(<ArticleControlBtn />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render ReportBtn', () => {
    const wrapper = shallow(<ReportBtn />);
    expect(wrapper).toMatchSnapshot();
  });
});
