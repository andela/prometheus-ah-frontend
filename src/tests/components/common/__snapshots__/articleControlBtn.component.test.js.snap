// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders component correctly should render ArticleControlBtn 1`] = `
<div>
  <Link
    className="dropdown-item"
    replace={false}
  >
    Edit
  </Link>
  <a
    className="dropdown-item"
    href="#"
    onClick={[Function]}
  >
    Delete
  </a>
</div>
`;

exports[`renders component correctly should render ReportBtn 1`] = `
<a
  className="dropdown-item"
  href="#"
>
  Report
</a>
`;
